neos/neos-ui
# BUGFIX: fix interdependent properties in node creation dialog

**What We did**
We added a transient _node_ and the _parentNode_ (based on _insertion mode_) to the _CreationDialog_ to be able to run basic `ClientEval:` operations known from the _Inspector_.
We also pulled the `prepocessing` of NodeConfigurations out of the _Inspector_ into a separate file.

We wrote tests for all the things we changed and added (unit & e2e).

**How I did it**
_Live_ with Neos community.

**How to verify it**
Add a Node with interdependent properties and creation dialog turned on. You can see an example NodeType added with this PR to the test setup.

Video:

https://user-images.githubusercontent.com/1615332/219017376-a03bb09a-0224-41a2-a4a4-d010d27db5d9.mp4

## TODO

- [x] get parentNode (depending on insert mode)
- [x] extract building transientNode and parentNode out of render method
- [x] build an abstraction for configuration processing to be used by both Insepector and CreationDialog
- [x] Docs / Comments


Resolves: #2852
